src/model/link.rs

```diff
@@ -1,9 +1,121 @@
-extern crate native_tls;
+extern crate rustls;
 
-use self::native_tls::{Certificate, TlsConnector, TlsStream};
 use model::data::Message;
 use model::error::{Error, RdpError, RdpErrorKind, RdpResult};
+use std::convert::TryInto;
 use std::io::{Cursor, Read, Write};
+use std::ops::{Deref, DerefMut};
+use std::sync::Arc;
+use std::time::SystemTime;
+
+use self::rustls::{
+    client::{NoClientSessionStorage, ServerCertVerified, ServerCertVerifier},
+    Certificate, ClientConfig, ClientConnection, ConnectionCommon, Error as RustlsError,
+    RootCertStore, ServerName, SideData, Stream as RustlsStream,
+};
+
+/// Marks all server certificates as valid
+/// so it can be used to turn off the server certificate
+/// validation on the client-side.
+struct DummyTlsVerifier;
+
+impl ServerCertVerifier for DummyTlsVerifier {
+    fn verify_server_cert(
+        &self,
+        _end_entity: &Certificate,
+        _intermediates: &[Certificate],
+        _server_name: &ServerName,
+        _scts: &mut dyn Iterator<Item = &[u8]>,
+        _ocsp_response: &[u8],
+        _now: SystemTime,
+    ) -> Result<ServerCertVerified, RustlsError> {
+        Ok(ServerCertVerified::assertion())
+    }
+}
+
+/// Copied from rustls library with removed trait bounds (Read + Write)
+/// By removing trait bounds we don't have to update
+/// other structs that depends on the Stream enum
+pub struct StreamOwned<C: Sized, T: Sized> {
+    /// Our conneciton
+    pub conn: C,
+
+    /// The underlying transport, like a socket
+    pub sock: T,
+}
+
+impl<C, T, S> StreamOwned<C, T>
+where
+    C: DerefMut + Deref<Target = ConnectionCommon<S>>,
+    T: Read + Write,
+    S: SideData,
+{
+    /// Make a new StreamOwned taking the Connection `conn` and socket-like
+    /// object `sock`.  This does not fail and does no IO.
+    ///
+    /// This is the same as `Stream::new` except `conn` and `sock` are
+    /// moved into the StreamOwned.
+    pub fn new(conn: C, sock: T) -> Self {
+        Self { conn, sock }
+    }
+
+    /// Get a reference to the underlying socket
+    pub fn get_ref(&self) -> &T {
+        &self.sock
+    }
+
+    /// Get a mutable reference to the underlying socket
+    pub fn get_mut(&mut self) -> &mut T {
+        &mut self.sock
+    }
+}
+
+impl<'a, C, T, S> StreamOwned<C, T>
+where
+    C: DerefMut + Deref<Target = ConnectionCommon<S>>,
+    T: Read + Write,
+    S: SideData,
+{
+    fn as_stream(&'a mut self) -> RustlsStream<'a, C, T> {
+        RustlsStream {
+            conn: &mut self.conn,
+            sock: &mut self.sock,
+        }
+    }
+}
```

Comment:
Is there a particular reason you wanted to include an explicit lifetime here?

If not, then you can move `as_stream` up to the previous block.

Reply:
Great catch! There is no reason for that explicit lifetime to be here anymore so I've moved it to the previous `impl` block as you've suggested.